lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -935,11 +935,33 @@ static Value getCommonOperand(Op op) {
 template <typename Op>
 static bool canonicalizeIdempotentInputs(Op op, PatternRewriter &rewriter) {
   auto inputs = op.getInputs();
-  llvm::SmallSetVector<Value, 8> uniqueInputs;
 
-  for (const auto input : inputs)
-    uniqueInputs.insert(input);
+  llvm::SmallSetVector<Value, 8> alreadyCovered;
+  llvm::SmallSetVector<Value, 8> checked;
```

Comment:
Since you only need `checked` to be a set, but you don't care about preserving the order in which elements were added:
```suggestion
  llvm::SmallDenseSet<Value, 8> checked;
```

Reply:
Thanks! :raised_hands: